File: tests/07_phases_ompss_2.dg/fortran/success_task_05.f90
! <testinfo>
! test_generator=config/mercurium-ompss-2
! test_nolink=yes
! </testinfo>
SUBROUTINE FOO(Y)
    IMPLICIT NONE
    INTEGER :: Y
END SUBROUTINE FOO

PROGRAM P
    IMPLICIT NONE
    INTEGER :: X
    INTERFACE
        !$OSS TASK VERIFIED INOUT(Y)
        SUBROUTINE FOO(Y)
            IMPLICIT NONE
            INTEGER :: Y
        END SUBROUTINE FOO
    END INTERFACE

    !$OSS TASK VERIFIED INOUT(X)
    !$OSS END TASK

    CALL FOO(X)

    !$OSS TASKWAIT
END PROGRAM P

File: tests/07_phases_ompss_2.dg/c/success_task_05.c
/*
<testinfo>
test_generator=config/mercurium-ompss-2
test_nolink=yes
</testinfo>
*/

#pragma oss task verified inout(*x)
void foo(int *x) {

}

int main() {
    int x;
    #pragma oss task verified inout(x)
    {}

    foo(&x);

    #pragma oss taskwait
}
